test/delegatemulti.js

```diff
@@ -0,0 +1,227 @@
+    );
+
+    for (let delegateTokenId of delegatees) {
+      let balance = await multiDelegate.balanceOf(deployer, delegateTokenId);
+      expect(balance.toString()).to.equal(
+        delegatorTokenAmount.div(delegatees.length).toString()
+      );
+    }
+
+    const newDelegatees = [bob, charlie];
+
+    const sourceTargetDelegateeArray = newDelegatees.map((newDelegatee, index) => [
+      delegatees[index],
+      newDelegatee
+    ]);
+
+    await multiDelegate.reDeposit(sourceTargetDelegateeArray);
+
+    for (let delegateTokenId of delegatees) {
+      let balance = await multiDelegate.balanceOf(deployer, delegateTokenId);
+      expect(balance.toString()).to.equal("0");
+    }
+
+    for (let delegateTokenId of newDelegatees) {
+      let balance = await multiDelegate.balanceOf(deployer, delegateTokenId);
+      expect(balance.toString()).to.equal(
+        delegatorTokenAmount.div(delegatees.length).toString()
+      );
+    }
+
+    // delegatee must have 1/2 of the votes the delegator delegated
+    const votesOfNewDelegatee = await token.getVotes(charlie);
+    expect(votesOfNewDelegatee.toString()).to.equal(
+      delegatorTokenAmount.div(newDelegatees.length).toString()
+    );
+  });
+
+  it('contract should revert if allowance is not provided', async () => {
+    const delegatorTokenAmount = await token.balanceOf(deployer);
+
+    const delegatees = [alice];
+    
+    const delegateeAmountArray = delegatees.map((delegatee) => ([
+      delegatee,
+      delegatorTokenAmount.div(delegatees.length)
+    ]));
+
+    await expect(
+      multiDelegate.depositMulti(
+        delegateeAmountArray
+      )
+    ).to.be.revertedWith('ERC20: insufficient allowance');
+  });
+
+  it('contract should revert if allowance is lesser than provided amount', async () => {
+    const delegatorTokenAmount = await token.balanceOf(deployer);
+    const customAmount = ethers.utils.parseEther('100000.0'); // total ens 77000000
+
+    // give allowance to multi delegate contract
+    await token.approve(multiDelegate.address, customAmount);
+
+    const delegatees = [bob];
+    const delegateeAmountArray = delegatees.map((delegatee) => ([
+      delegatee,
+      delegatorTokenAmount.div(delegatees.length)
+    ]));
+    await expect(
+      multiDelegate.depositMulti(delegateeAmountArray)
+    ).to.be.revertedWith('ERC20: insufficient allowance');
+  });
+
+  it('contract should revert if no delegatee provided', async () => {
+    const delegatorTokenAmount = await token.balanceOf(deployer);
+
+    // give allowance to multi delegate contract
+    await token.approve(multiDelegate.address, delegatorTokenAmount);
+
+    const delegatees = [];
+    const delegateeAmountArray = delegatees.map((delegatee) => ([
+      delegatee,
+      delegatorTokenAmount.div(delegatees.length)
+    ]));
+    await expect(
+      multiDelegate.depositMulti(delegateeAmountArray)
+    ).to.be.revertedWith('You should pick at least one delegatee');
+  });
+});
```

Comment:
This needs to add tests for delegting to and transferring to an account that already has a proxy contract.